apps/emqx_connector/test/emqx_connector_jwt_worker_SUITE.erl

```diff
@@ -0,0 +1,340 @@
+        {'PrivateKeyInfo', <<"xxxxxx">>, not_encrypted},
+        {'PrivateKeyInfo', <<"xxxxxx">>, not_encrypted}
+    ]),
+    Config = Config0#{private_key := InvalidPEM},
+    process_flag(trap_exit, true),
+    ?check_trace(
+        ?wait_async_action(
+            ?assertMatch({ok, _}, emqx_connector_jwt_worker:start_link(Config)),
+            #{?snk_kind := connector_jwt_worker_startup_error},
+            1_000
+        ),
+        fun(Trace) ->
+            ?assertMatch(
+                [#{error := {invalid_private_key, _}}],
+                ?of_kind(connector_jwt_worker_startup_error, Trace)
+            ),
+            ok
+        end
+    ),
+    ok.
+
+t_refresh(_Config) ->
+    Config0 =
+        #{
+            table := Table,
+            resource_id := ResourceId
+        } = generate_config(),
+    Config = Config0#{expiration => 5_000},
+    ?check_trace(
+        begin
+            {{ok, _Pid}, {ok, _Event}} =
+                ?wait_async_action(
+                    emqx_connector_jwt_worker:start_link(Config),
+                    #{?snk_kind := connector_jwt_worker_token_stored},
+                    5_000
+                ),
+            {ok, FirstJWT} = emqx_connector_jwt:lookup_jwt(Table, ResourceId),
+            ?block_until(
+                #{
+                    ?snk_kind := connector_jwt_worker_refresh,
+                    jwt := JWT0
+                } when JWT0 =/= FirstJWT,
+                15_000
+            ),
+            {ok, SecondJWT} = emqx_connector_jwt:lookup_jwt(Table, ResourceId),
+            ?assertNot(is_expired(SecondJWT)),
+            ?assert(is_expired(FirstJWT)),
+            {FirstJWT, SecondJWT}
+        end,
+        fun({FirstJWT, SecondJWT}, Trace) ->
+            ?assertMatch(
+                [_, _ | _],
+                ?of_kind(connector_jwt_worker_token_stored, Trace)
+            ),
+            ?assertNotEqual(FirstJWT, SecondJWT),
+            ok
+        end
+    ),
+    ok.
+
+t_format_status(_Config) ->
+    Config = generate_config(),
+    {ok, Pid} = emqx_connector_jwt_worker:start_link(Config),
+    {status, _, _, Props} = sys:get_status(Pid),
+    [State] = [
+        State
+     || Info = [_ | _] <- Props,
+        {data, Data = [_ | _]} <- Info,
+        {"State", State} <- Data
+    ],
+    ?assertMatch(
+        #{
+            jwt := "******",
+            jwk := "******"
+        },
+        State
+    ),
+    ok.
+
+t_lookup_ok(_Config) ->
+    Config =
+        #{
+            table := Table,
+            resource_id := ResourceId,
+            private_key := PrivateKeyPEM,
+            aud := Aud,
+            iss := Iss,
+            sub := Sub,
+            kid := KId
+        } = generate_config(),
+    {ok, Worker} = emqx_connector_jwt_worker:start_link(Config),
+    Ref = emqx_connector_jwt_worker:ensure_jwt(Worker),
+    receive
+        {Ref, token_created} ->
+            ok
+    after 500 ->
+        error(timeout)
```

Comment:
NIT: `ct:fail`?

Reply:
Yes, `ct:fail`'s only advantage is declaring intentions in the code explicitly :)